### PR TITLE
chore: convert digital land API files from JS to TS

### PR DESCRIPTION
**Pull request by jessicamcinchak**
relies on https://github.com/theopensystemslab/planx-core/pull/37

first step of improving our GIS API. lots of room for future refactors of this big function into smaller, more testable methods next.

this example endpoint should return 200 & be formatted consistently with prod: 
- https://api.1733.planx.pizza/gis/opensystemslab?geom=POLYGON+%28%28-0.07629275321961124+51.48596289289142%2C+-0.0763061642646857+51.48591028066045%2C+-0.07555112242699404+51.48584764697301%2C+-0.07554173469544191+51.48590192950712%2C+-0.07629275321961124+51.48596289289142%29%29&analytics=false

### CODE REVIEW

**Review by builditben — APPROVED** [review]
The response format is slightly different to staging with some minor additions and changes (and the example query provided returns very different data in prod but I assume that is because of other work pending deployment). The constraints component is still working as expected. :+1:

File: api.planx.uk/gis/digitalLand.ts [hunk 1]
```diff
@@ -83,23 +82,26 @@ async function go(localAuthority, geom, extras) {
 
   // --- INTERSECTIONS ---
   // check for & add any 'positive' constraints to the formattedResult
-  let formattedResult = {};
+  let formattedResult: Record<string, Constraint> = {};
   if (res && res.count > 0 && res.entities) {
-    res.entities.forEach((entity) => {
+    res.entities.forEach((entity: { dataset: any; }) => {
```
[the remark this answers is not in the document]
Comment (DafyddLlyr) [reply api.planx.uk/gis/digitalLand.ts:87]: Just seen https://github.com/theopensystemslab/planx-core/pull/37 which has this as a WIP - that's great

File: api.planx.uk/gis/digitalLand.ts [hunk 2]
```diff
@@ -192,24 +196,24 @@ async function go(localAuthority, geom, extras) {
...
       // get the planx variable that corresponds to this 'dataset', should never be null because we only requested known datasets
-      const key = Object.keys(baseSchema).find((key) => baseSchema[key]["digital-land-datasets"].includes(response.dataset));
-      metadata[key] = response;
+      const key = Object.keys(baseSchema).find((key) => baseSchema[key]["digital-land-datasets"]?.includes(response.dataset));
+      if (key) metadata[key] = response;
     });
   }).catch(error => console.log(error));
 
-  return { url: url, constraints: formattedResult, metadata: metadata };
+  return { url: url, constraints: formattedResult, metadata: metadata } as GISResponse;
```
Comment (DafyddLlyr) [inline api.planx.uk/gis/digitalLand.ts:213]: nit: We can avoid casting here by setting the return type of `go()` to be `Promise<GISResponse>`

**Review by DafyddLlyr — APPROVED** [review]

File: api.planx.uk/gis/digitalLand.ts [hunk 3]
```diff
@@ -1,10 +1,12 @@
...
+import type { Constraint, GISResponse, Metadata } from "@opensystemslab/planx-core/types";
 import { gql } from "graphql-request";
+import fetch from "isomorphic-fetch";
+import { adminGraphQLClient as adminClient } from "../hasura";
+import { addDesignatedVariable, omitGeometry } from "./helpers";
+import { baseSchema } from "./local_authorities/metadata/base";
 
-const localAuthorityMetadata = {
+
+const localAuthorityMetadata: Record<string, any> = {
```
[the remark this answers is not in the document]
Comment (jessicamcinchak) [reply api.planx.uk/gis/digitalLand.ts:9]: Right, thanks for spotting this - it's a good reminder to also convert a few of the council-specific metadata files to .ts as well in addition as that's where the A4s are configured :+1: I'll update this branch with those shortly

File: api.planx.uk/gis/digitalLand.ts (hunk 2 above)
Comment (jessicamcinchak) [reply api.planx.uk/gis/digitalLand.ts:220]: of course, will update :+1:

**Pull request merged**